src/main/java/net/mcreator/ui/vcs/MCreatorWorkspaceSyncHandler.java

```diff
@@ -45,13 +45,7 @@
 import java.util.concurrent.ConcurrentHashMap;
 import java.util.stream.Collectors;
 
-public class MCreatorWorkspaceSyncHandler implements ICustomSyncHandler {
-
-	private final MCreator mcreator;
-
-	public MCreatorWorkspaceSyncHandler(MCreator mcreator) {
-		this.mcreator = mcreator;
-	}
+public record MCreatorWorkspaceSyncHandler(MCreator mcreator) implements ICustomSyncHandler {

```

Comment:
Record is meant for "data classes", not for classes that do actual logic, this class is not suitable for record from semantic point of view

Reply:
I figured that would shorten the code by few lines. It could suppress compiler warnings as well if you want:
```java
@SuppressWarnings("ClassCanBeRecord")
```